### PR TITLE
Eta phi search in tile

### PR DESCRIPTION
**Pull request by rovere**
#### PR description:

The method `searchBoxEtaPhi` of the Tile data structure returns the
correct bin numbers in eta and phi, but ignores the possible wrapping
problems that could happen when the phi range is crossing the
"bin-boundary". Therefore, a trivial loop on the client-side between the
returned eta and phi bin ranges could result in fewer cells explored.
The solution implemented will return proper intervals for both eta and
phi upon which the caller can perform a double for loop. It is the
caller's responsibility, though, to wrap the phi-bin inside the phi-loop
to get the correctly wrapped global bin value.
A corresponding unit test has been added to verify the correctness of
the implementation.

#### PR validation:

Tested on a Phase2 workflow with several clusters in the Scintillator section with some 1K events. No regression was observed.

### CODE REVIEW

**Comment by rovere** [comment]
@cmsbuild please test

**Comment by jpata** [comment]
enable profiling

**Comment by jpata** [comment]
@cmsbuild please test

**Comment by jpata** [comment]
+reconstruction
- technical improvement to phi wrapping in HGCAL reconstruction
- no reco/DQM changes (none expected as I understand from the description)
- profiling does not pick up any relevant changes
- unit tests for the new functionality were added (thanks!)

**Comment by srimanob** [comment]
+Upgrade

File: RecoLocalCalo/HGCalRecProducers/interface/HGCalLayerTiles.h [hunk 1]
```diff
@@ -66,11 +60,9 @@ class HGCalLayerTilesT {
   }
 
   int getPhiBin(float phi) const {
```
Comment (perrotta) [inline RecoLocalCalo/HGCalRecProducers/interface/HGCalLayerTiles.h:62]: The method `getPhiBin` becomes now identical to `TICLLayerTile::phiBin`

Analogously, `getEtaBin` in this class is the same as `TICLLayerTile::etaBin,` and the same for `getGlobalBinByBin` and `getGlobalBinEtaPhi`.

(Moreovers, it is recommended to not start getter methods with "get": as such the names of the methods as in TICLLayerTile must be preferred.

Probably taking them from a common class could help avoid having to fix any of them independently from the others

**Comment by perrotta** [comment]
+1
- The fix looks correct
- However, avoiding duplicating code among similar classes could help maintaining it in both classes: this should be taken into account in possible future updates and code cleaning campaigns

**Comment by jpata** [comment]
type hgcal

**Comment by rovere** [comment]
@perrotta thanks for the approval and for the comments. On the other hand, the 2 places where the fix to the eta-phi search has been deployed cover different use-cases and cannot be easily merged. One Tile structure is fully projective in `eta-phi`, while the other is hosting both `x-y` and `eta-phi` partitioning in a unique place. The former is used during the pattern recognition, the latter while creating the `LayerClusters` that need special treatment for the scintillator part (hence the mixed space partitioning). It is not foreseen, at least in the near future, to merge the 2 together.

**Pull request merged**